### PR TITLE
added new interface for simple header

### PR DESCRIPTION
**Pull request by sasurobert**
added new interface for simple header

### CODE REVIEW

File: data/interface.go [hunk 1]
```diff
@@ -75,14 +66,36 @@ type HeaderHandler interface {
...
+	SetDeveloperFees(value *big.Int) error
+	SetAccumulatedFees(value *big.Int) error
 	SetMiniBlockHeaderHandlers(mbHeaderHandlers []MiniBlockHeaderHandler) error
 	SetReceiptsHash(hash []byte) error
-	SetScheduledRootHash(rootHash []byte) error
 	ValidateHeaderVersion() error
+	IsInterfaceNil() bool
+}
+
+// ValidatorStatisticsInfoHandler simple handler needed for validator info
```
Comment (SebastianMarian) [inline data/interface.go:77]: is a simple...

File: data/interface.go [hunk 2]
```diff
@@ -75,14 +66,36 @@ type HeaderHandler interface {
...
+type ValidatorStatisticsInfoHandler interface {
+	SetValidatorStatsRootHash(rHash []byte) error
```
Comment (SebastianMarian) [inline data/interface.go:79]: rootHash

File: .github/workflows/pr-tests.yml [hunk 3]
```diff
@@ -11,14 +11,14 @@ jobs:
     name: Unit
     runs-on: ubuntu-latest
     steps:
-      - name: Set up Go 1.x
-        uses: actions/setup-go@v2
+      - name: Set up Go 1.17.6
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply .github/workflows/pr-tests.yml:14]: this is a merge from master.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Binaries for programs and plugins
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply .gitignore:1]: this is a merge from master.

File: LICENSE [hunk 5]
```diff
@@ -0,0 +1,674 @@
+                    GNU GENERAL PUBLIC LICENSE
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply LICENSE:1]: this is a merge from master.

File: README.md [hunk 6]
```diff
@@ -0,0 +1,2 @@
+# elrond-go-core
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply README.md:1]: this is a merge from master.

**Review by iulianpascalau — COMMENTED** [review]
Only what Sebi found

File: data/block/headerWithValidatorStats.go [hunk 7]
```diff
@@ -1,2 +1,451 @@
...
+
+	return hv.Header.SetTxCount(txCount)
+}
+
+// SetShardID sets header shard ID
+func (hv *HeaderWithValidatorStats) SetShardID(_ uint32) error {
+	return nil
+}
+
+// GetMiniBlockHeadersWithDst as a map of hashes and sender IDs
```
Comment (SebastianMarian) [inline data/block/headerWithValidatorStats.go:354]: Reformulate comment

File: data/block/headerWithValidatorStats.go [hunk 8]
```diff
@@ -1,2 +1,451 @@
...
+// SetReceiptsHash sets the receipts hash
+func (hv *HeaderWithValidatorStats) SetReceiptsHash(hash []byte) error {
+	if hv == nil {
+		return data.ErrNilPointerReceiver
+	}
+
+	return hv.Header.SetReceiptsHash(hash)
+}
+
+// HasScheduledSupport returns true as the second block version does support scheduled data
```
Comment (SebastianMarian) [inline data/block/headerWithValidatorStats.go:450]: Reformulate comment

File: data/block/headerWithValidatorStats.go [hunk 9]
```diff
@@ -340,6 +340,17 @@ func (hv *HeaderWithValidatorStats) SetShardID(_ uint32) error {
 	return nil
 }
 
+// SetValidatorStatsRootHash sets the root hash for the validator statistics trie
+func (hv *HeaderWithValidatorStats) SetValidatorStatsRootHash(rHash []byte) error {
```
Comment (SebastianMarian) [inline data/block/headerWithValidatorStats.go:344]: rootHash

Comment (sasurobert) [reply data/block/headerWithValidatorStats.go:379]: done

File: data/block/headerWithValidatorStats.go (hunk 8 above)
Comment (sasurobert) [reply data/block/headerWithValidatorStats.go:490]: done

File: data/block/headerWithValidatorStats.go (hunk 7 above)
Comment (sasurobert) [reply data/block/headerWithValidatorStats.go:389]: done

File: data/interface.go (hunk 1 above)
Comment (sasurobert) [reply data/interface.go:81]: this does not exists

File: data/block/headerWithValidatorStats.go (hunk 9 above)
Comment (sasurobert) [reply data/block/headerWithValidatorStats.go:379]: This is the same everywhere - this is legacy. You would need to change a lot on chain-go repo when integrating such change

File: data/block/block.go [hunk 10]
```diff
@@ -416,8 +445,8 @@ func (b *Body) IsInterfaceNil() bool {
 	return b == nil
 }
 
-// Clone the underlying data
-func (mb *MiniBlock) Clone() *MiniBlock {
+// DeepClone the underlying data
+func (mb *MiniBlock) DeepClone() *MiniBlock {
```
Comment (iulianpascalau) [inline data/block/block.go:449]: :+1: yup, this is deep clone actually

**Review by iulianpascalau — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**